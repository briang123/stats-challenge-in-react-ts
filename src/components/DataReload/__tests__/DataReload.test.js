import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { DATASET_1, DATASET_2 } from './../../../common/constants';
import DataReload from '../DataReload';

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  cleanup();
});

describe('<DataReload />', () => {
  it('Should not render the reload container if we are not supplying any props', () => {
    const { queryByText } = render(<DataReload />);

    const container = queryByText('reload-container');
    expect(container).toBeNull();
  });

  it('Should not render the reload container if we are passing an invalid onClick function as the prop', () => {
    const { queryByText } = render(<DataReload onClick="" />);

    const container = queryByText('reload-container');
    expect(container).toBeNull();
  });

  it('Should match snapshot without props', () => {
    const tree = create(<DataReload />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render the reload container buttons if a valid onClick function is passed', () => {
    const validFunction = jest.fn();
    const { getByTestId } = render(<DataReload onClick={validFunction} />);

    const container = getByTestId('reload-container');
    expect(container.tagName).toBe('DIV');
    expect(container.childElementCount).toBe(1);

    const btnContainer = container.firstChild;
    expect(btnContainer.tagName).toBe('DIV');

    expect(btnContainer.childElementCount).toBe(2);

    const reload1234Btn = getByTestId(`btn-${DATASET_1.onClickArg}`);
    expect(reload1234Btn.tagName).toBe('BUTTON');
    expect(reload1234Btn.textContent).toBe(DATASET_1.buttonText);

    const reload4321Btn = getByTestId(`btn-${DATASET_2.onClickArg}`);
    expect(reload4321Btn.tagName).toBe('BUTTON');
    expect(reload4321Btn.textContent).toBe(DATASET_2.buttonText);
  });

  it('Button click events should pass when passing in an onClick event handler', () => {
    const validFunction = jest.fn();
    const { getByTestId } = render(<DataReload onClick={validFunction} />);

    const reload1234Btn = getByTestId(`btn-${DATASET_1.onClickArg}`);
    fireEvent.click(reload1234Btn);
    expect(validFunction).toHaveBeenCalled();
    expect(console.error).toHaveBeenCalledTimes(0);

    const reload4321Btn = getByTestId(`btn-${DATASET_2.onClickArg}`);
    fireEvent.click(reload4321Btn);
    expect(validFunction).toHaveBeenCalled();
    expect(console.error).toHaveBeenCalledTimes(0);
  });

  it('Should match snapshot with props', () => {
    const validFunction = jest.fn();
    const tree = create(<DataReload onClick={validFunction} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
