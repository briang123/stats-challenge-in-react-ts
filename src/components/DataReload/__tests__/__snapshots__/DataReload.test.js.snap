// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataReload /> Should match snapshot with props 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: none;
  height: 4.87rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: var(--lightcolor);
  outline: none;
  border-radius: 44px;
  width: 375px;
  background-color: var(--defaultbtn);
}

.c2:hover {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100px;
  min-width: 100vw;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  background-color: var(--secondarybg);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-width: 50vw;
  height: 125px;
}

.c1 button {
  margin: 10px;
}

@media (min-width:600px) {
  .c1 {
    min-width: 55vw;
  }

  .c1 button {
    margin: 0px;
  }
}

<div
  className="c0"
  data-testid="reload-container"
>
  <div
    className="c1"
  >
    <button
      className="c2"
      data-testid="btn-1234"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
    >
      Reload JSON-1234 Data
    </button>
    <button
      className="c2"
      data-testid="btn-4321"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
    >
      Reload JSON-4321 Data
    </button>
  </div>
</div>
`;

exports[`<DataReload /> Should match snapshot without props 1`] = `null`;
